pandas/io/pytables.py

```diff
@@ -593,13 +593,20 @@ def __exit__(self, exc_type, exc_value, traceback):
     def keys(self) -> List[str]:
         """
         Return a list of keys corresponding to objects stored in HDFStore.
+        If the store contains pandas native tables, it will return their names.
```

Comment:
this is not a good api.

either:
a) always return all tables (api breaking), or 
b) add a keyword include='pandas'|'native'|'all', default would be 'pandas'

Reply:
Note I am not the PR author but reported the issue.

It really depends on what you want to achieve, IMHO better compatibility would be achieved with option (a) which basically reverts #21543 but for `keys()` and leaving `groups()` untouched. Don't know if that breaks anything else, but it might hurt performance in some cases although the original bug  #21372 complained about the performance of `groups()` (it links to #17593 about slow `keys()`). There is also a `walk()` method which is very similar to `groups()`.

(b) would break backwards compatibility of all (future) projects that use that keyword wrt `pandas < 1.0.x`.

Reply:
The latest version is trying to keep some backwards compatibility with pandas 0.23 (plus or minus a minor revision). Apparently there are some tools that depend on the original behavior (hence the issue which this PR is trying to fix). So either we keep breaking backward compatibility or we add a keyword.
I think this is a matter of taste/policy.

Reply:
@jreback I think this is a trade-off between having backwards compatible behavior with v0.23.x vs. a clean API.
I see one vote from @st-bender  for being backwards compatible, and I see one vote from @jreback for the clean API
Note sure whose vote has more weight (:smile: but I can make a gues), but I have no strong feelings either way.

Reply:
Thanks @roberthdevries for pushing this forward and @jreback thanks for the feedback.
As I see it there is no fully compatible way to change that behaviour back to <=0.23.x, and apparently I am the only one that noticed that when `dask.dataframe.read_hdf()` suddenly failed to read some files which worked before (dask/dask#5934).

It seems that `dask.dataframe` is currently the only serious "user" of that interface, and I filed dask/dask#5934 to decide whether it is better handled within `dask` instead of `pandas`. The change there would be similarly simple and would probably be more backwards compatible with older pandas versions.